### ISSUE TITLE
Missing transaction on SPI read() / write() mocks in 1-alpha

### ISSUE THREAD

**Issue by teotwaki** [dbrgn/embedded-hal-mock#71]
Hi,

It appears to me that there are missing transactions on the `SpiDeviceWrite::write()` and `SpiDeviceRead::read()` functions. The 1.0.0-alpha.10 documentation for `SpiDeviceWrite::write()` [indicates the following](https://docs.rs/embedded-hal/1.0.0-alpha.10/embedded_hal/spi/trait.SpiDeviceWrite.html#method.write):

> Do a write within a transaction.
> 
> This is a convenience method equivalent to device.write_transaction(&mut [buf]).

However, when I try to mock this call to `write()` away, I get the following test failure:

```
---- tests::write_reg_transforms_addr stdout ----
thread 'tests::write_reg_transforms_addr' panicked at 'assertion failed: `(left == right)`
  left: `TransactionStart`,
 right: `Write`: spi::write unexpected mode', /Users/slau/.cargo/git/checkouts/embedded-hal-mock-a654aa1079912b96/33b0458/src/spi.rs:286:9
 ```

This issue did not exist for 1.0.0-alpha.9 as far as I can tell.

**Comment by Artur-Romaniuk** [comment]
@teotwaki Can this issue and PR be closed now, as the `embedded-hal = "1.0.0-rc.3"` doesn't require write and read implementations as they are already provided?

**Comment by teotwaki** [comment]
Of course.
